### PR TITLE
Use ipfs-core-types; Updated npm git dependency format; Minor fixes

### PR DESCRIPTION
**Pull request by ple1n**
Changes include https://github.com/orbitdb/orbit-db-types/pull/31

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
   "dependencies": {
     "@types/elliptic": "^6.4.6",
     "@types/events": "^1.2.0",
-    "@types/ipfs": "git+https://github.com/lukas2005/types-ipfs.git",
-    "orbit-db": "git+https://github.com/orbitdb/orbit-db.git"
+    "ipfs-core-types": "^0.10.1",
+    "orbit-db": "orbitdb/orbit-db"
```
Comment (tabcat) [inline package.json:25]: how does this work? is this abbreviated github?

**Review by tabcat — COMMENTED** [review]
thanks for making these changes @planetoryd 
will be good to use the official ipfs types

File: package.json (hunk 1 above)
Comment (ple1n) [reply package.json:25]: https://docs.npmjs.com/cli/v8/configuring-npm/package-json#github-urls

both should work i think. for some reason i changed it to this (git+url seemed slower)

File: Store.d.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ declare module "orbit-db-store" {
         /** 
          * Contains all entries of this Store
          */
-        all: any[];
+        all: any[] | {};
```
Comment (tabcat) [inline Store.d.ts:18]: this should only be `any[]`

https://github.com/orbitdb/orbit-db-store/blob/6a450aca4ba7f5a041677f5cf25632f0170cf7b7/src/Store.js#L155

Comment (ple1n) [reply Store.d.ts:18]: It should be, but the IDE told me it could be `{}` too, for some reason. uh, right, the implementation should be changed instead

Comment (tabcat) [reply Store.d.ts:18]: I dont think the code block I linked needs to be changed, and im not sure how it could return an object.

**Comment by ple1n** [comment]
Sorry I'm not interested in using orbit-db as a dependency anymore. It seems rewriting that part would be better

File: Store.d.ts (hunk 2 above)
Comment (ple1n) [reply Store.d.ts:18]: Yes it shouldn't be. If for some reason it returns an object it's the fault of orbit-db not its typing.

**Pull request closed without merging**